src/tests/ngraph_helpers/snippets_ngraph_functions/include/precision_propagation_function.hpp

```diff
@@ -0,0 +1,159 @@
+        return model;
+    }
+
+private:
+    static std::shared_ptr<Node> create_convert(std::shared_ptr<Node> parent, const element::Type convertion_type) {
+        return convertion_type == element::undefined
+                   ? std::dynamic_pointer_cast<Node>(parent)
+                   : std::make_shared<ngraph::snippets::op::ConvertSaturation>(parent, convertion_type);
+    }
+
+    static std::pair<std::shared_ptr<ngraph::opset1::Parameter>, std::shared_ptr<ov::Node>> make_branch(
+        const ngraph::element::Type precision,
+        const ngraph::PartialShape& inputShape,
+        const size_t index,
+        const element::Type convertion_type) {
+        const auto parameter = std::make_shared<ngraph::opset1::Parameter>(precision, inputShape);
+        parameter->set_friendly_name("parameter" + std::to_string(index));
+
+        std::shared_ptr<Node> parent = create_convert(parameter, convertion_type);
+
+        return {parameter, parent};
+    }
+};
+
+class PrecisionPropagationFunctionParams {
+public:
+    class Actual {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+    };
+
+    class Expected {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+        element::Type convertion_after_op2;
+    };
+};
+
+template<typename T>
+class EmpyInfrastructureSupportStubForPrecisionPropagationFunction : public SnippetsFunctionBase {
+public:
+    explicit EmpyInfrastructureSupportStubForPrecisionPropagationFunction(
+        const std::vector<PartialShape> input_shapes,
+        const ngraph::element::Type precision1,
+        const ngraph::element::Type precision2,
+        const ngraph::element::Type constant_precision,
+        PrecisionPropagationFunctionParams::Actual actual,
+        PrecisionPropagationFunctionParams::Expected expected) :
+        SnippetsFunctionBase(input_shapes),
+        precision1(precision1),
+        precision2(precision2),
+        constant_precision(constant_precision),
+        actual(actual),
+        expected(expected) {
+        OPENVINO_ASSERT(input_shapes.size() == 2ull, "input_shapes size has to be equal to 2");
+    }
+
+protected:
+    std::shared_ptr<ov::Model> initOriginal() const override {
+        return PrecisionPropagationFunction::get<T>(
+            precision1,
+            input_shapes[0],
+            precision2,
+            input_shapes[1],
+            constant_precision,
+            actual.convertion_before_op1,
+            actual.convertion_before_op2_1,
+            actual.convertion_before_op2_2);
+    }
+
+    std::shared_ptr<ov::Model> initReference() const override {
```

Comment:
Why would you override different methods to yield the same function. Please, take a look at other ngraph snippets functions implementations [here](https://github.com/openvinotoolkit/openvino/blob/fb3d785c0662c4c0c5864162d96c0ee09b220f45/src/tests/ngraph_helpers/snippets_ngraph_functions/include/subgraph_simple.hpp#L168-L229), they all override only `initOriginal()` since it's enough for functional tests